components/sidebar/SidebarNoteLink.tsx

```diff
@@ -81,7 +81,10 @@ const SidebarNoteLink = (
             fill="currentColor"
           />
         </button>
-        <span className="overflow-hidden overflow-ellipsis whitespace-nowrap">
+        <span
+          data-testid={'sidebar-note-link-' + note.title}
```

Comment:
Instead of adding a `data-testid` here, can we get this element by using `contains`?